Sourcery/DecodableWithDefault.swift.stencil

```diff
@@ -1,16 +1,28 @@
 {% for type in types.all|annotated:"decodableWithDefault" %}
 extension {{ type.name }}: Decodable {
-    enum CodingKeys: String, CodingKey {
+    enum KnownKeys: String {
         {% for variable in type.variables|stored %}
         case {{ variable.name }}
         {% endfor %}
     }
 
     public init(from decoder: Decoder) throws {
-        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let container = try StringCodingContainer<KnownKeys>(decoder: decoder)
         {% for variable in type.variables|stored %}
-        {{ variable.name }} = try container.decodeIfPresent({{ variable.typeName|replace:"?","" }}.self, forKey: .{{ variable.name }}) ?? {{ variable.defaultValue }}
+
+        {{ variable.name }} = try container.decode({{ variable.typeName }}.self,
+            forKey: .{{ variable.name }},
+            defaultValue: {{ variable.defaultValue }}
+        )
+        {% endfor %}
+
+        container.recordPotentialIssues(deprecations: [
+        {% for variable in type.variables|stored %}
+        {% if variable.annotations.deprecated %}
+            (.{{ variable.name }}, "{{ variable.annotations.message }}"),
+        {% endif %}
         {% endfor %}
```

Comment:
It might not be quite clear from this template, especially since we're not using it yet, but this allows us to annotate properties in **ConfigOptions.swift** as deprecated so that they produce warnings.

For example:

```swift
// sourcery: skip, deprecated, message = "Renamed to overriddenResponses"
public var overridenResponses: [String: String] = [:]
```

The annotations:

- `skip` - Omits the property from the documentation
- `deprecated` - Causes it to be included in this `deprecations` array
- `message` - Extra details to be printed for the user helping to guide them away from this property